homeassistant/components/verisure/coordinator.py

```diff
@@ -83,13 +83,16 @@ async def _async_update_data(self) -> dict:
             raise UpdateFailed("Could not read overview") from err
 
         def unpack(overview: list, value: str) -> dict | list:
-            return next(
-                (
-                    item["data"]["installation"][value]
-                    for item in overview
-                    if value in item.get("data", {}).get("installation", {})
-                ),
-                [],
+            return (
+                next(
+                    (
+                        item["data"]["installation"][value]
+                        for item in overview
+                        if value in item.get("data", {}).get("installation", {})
+                    ),
+                    [],
```

Comment:
We could save creating this list and just put `None` instead since we're going to create an additional list anyway as default below.

Reply:
Problem was that `unpack(overview, "cameras")` returned `None` if the overview would contain something like `[ {"data": {"installation": {"cameras": None } } } ]`.

I originally thought the default in the 'next' function would have caught it. That only works if the input to the next function is `None`, not if it's `(None, )`.

The list comprehension `device["device"]["deviceLabel"]: device for device in unpack(overview, "cameras")` would then raise a `TypeError: 'NoneType' object is not iterable`.

Reply:
Yes, but an empty list is also falsy so the `or` statement will be evaluated.

Reply:
Yes, that's the whole point of this PR?

Reply:
without the empty list as default in the `next` function, it would raise `StopIteration`

Reply:
I'm suggesting replacing the empty list with `None` in `next` second parameter.